src/test/java/mate/academy/bookstore/controller/OrderControllerTest.java

```diff
@@ -0,0 +1,181 @@
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.com.github.dockerjava.core.MediaType;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class OrderControllerTest {
+    private static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext
+    ) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(SecurityMockMvcConfigurers.springSecurity())
+                .build();
+    }
+
+    @WithMockUser(username = "user")
+    @Test
+    @DisplayName("Verify placeAnOrder() method works for OrderController")
+    @Sql(scripts = {
+            "classpath:database/order/create-shopping-cart.sql"
+    }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {
+            "classpath:database/order/remove-shopping-cart.sql"
+    }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void placeAnOrder_ValidRequestDto_ReturnOrderDto() throws Exception {
+        OrderDto expected = new OrderDto(
+                1L,
+                17L,
+                Set.of(new OrderItemDto(1L, 1L, 1)),
+                null,
+                BigDecimal.valueOf(99.99),
+                Order.Status.PENDING
+        );
+        PlaceAnOrderDto requestDto = new PlaceAnOrderDto("SomeAddress");
+        Authentication authentication = authentication();
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(
+                        MockMvcRequestBuilders.post("/orders")
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON.getMediaType())
+                                .with(SecurityMockMvcRequestPostProcessors
+                                        .authentication(authentication))
+                )
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andReturn();
+
+        OrderDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), OrderDto.class
+        );
+        EqualsBuilder.reflectionEquals(expected, actual, "id", "orderDate");
+    }
+
+    @WithMockUser(username = "user")
+    @Test
+    @DisplayName("Verify getOrders() method works for OrderController")
+    @Sql(scripts = {
+            "classpath:database/order/create-order.sql"
+    }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {
+            "classpath:database/order/remove-order.sql"
+    }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void getOrders_ValidRequestDto_ReturnListOrderDto() throws Exception {
+        OrderDto expected = new OrderDto(
```

Comment:
Code duplication. To remove creation of same test data I suggest to create constants for common data or create method or even separate class to for such purpose